hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -342,11 +349,12 @@ object Simplify {
           mct,
           Subst(pred, Env.empty[IR].bind("sa" -> Ref("row", mct.typ.rowType))))
       case MatrixColsTable(MatrixMapGlobals(child, newRow)) => TableMapGlobals(MatrixColsTable(child), newRow)
-      case MatrixColsTable(MatrixMapRows(child, _, _)) => MatrixColsTable(child)
+      case MatrixColsTable(MatrixMapRows(child, _)) => MatrixColsTable(child)
       case MatrixColsTable(MatrixMapEntries(child, _)) => MatrixColsTable(child)
       case MatrixColsTable(MatrixFilterEntries(child, _)) => MatrixColsTable(child)
       case MatrixColsTable(MatrixFilterRows(child, _)) => MatrixColsTable(child)
       case MatrixColsTable(MatrixAggregateRowsByKey(child, _)) => MatrixColsTable(child)
+      case MatrixColsTable(MatrixKeyRowsBy(child, _, _)) => MatrixColsTable(child)
```

Comment:
we need an additional rule to invert `MatrixRowsTable(MatrixKeyRowsBy(` to `TableKeyBy(MatrixRowsTable(`.

Reply:
Already did! Line 336.